### PR TITLE
fix: service apply, update, and env commands handle edge cases

### PR DESCRIPTION
**Pull request by amaechiabuah**
## Describe Changes

Three related fixes in the service resource:

- **`find`**: The V3 `replicationcontroller/{name}` endpoint returns HTTP 200 with a null body for non-existent services. This caused `apply` to take the update path instead of create. Now checks for a null/empty response and raises `DuploNotFound`.
- **`update`**: Crashes with `KeyError: 'Template'` when given a flat YAML body (e.g. from `apply`). Now uses `body.get("Template", body)` to handle both API response format (nested Template) and user YAML (flat).
- **`update_env` / `update_labels`**: Crash with `JSONDecodeError` when `OtherDockerConfig` is empty string. Now defaults to `"{}"`.

## Link to Issues

https://app.clickup.com/t/8655600/DUPLO-41899

## PR Review Checklist

- [x] Thoroughly reviewed on local machine.
- [ ] Have you added any tests
- [x] Make sure to note changes in Changelog

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-41899 Service: update_env fails](https://app.clickup.com/t/8655600/DUPLO-41899)

**Comment by Matheusbafutto** [comment]
Even though you explained the fixes done, for the case of update from flat YAML I am still having trouble visualizing what was actually fixed.

If possible I would paste the payload used for testing in the description/comment OR adding unit tests covering the edge cases this is solving for (both would improve reviewer visibility)

**Comment by amaechiabuah** [comment]
Added unit tests in 0b34eb5 covering each edge case. Here are the payloads that trigger them:
                                                                                                                               
  1. find() — V3 endpoint returns 200 with null body
                                                                                                                               
  The V3 replicationcontroller/{name} endpoint returns HTTP 200 with a null JSON body for non-existent services. Before the fix, apply would see a truthy response and take the update path instead of create.                                           
                                                                                                                               
  ```
// GET v3/subscriptions/{tenant_id}/replicationcontroller/nonexistent-svc
  // HTTP 200                                                                                                                  
  null    
```                                                                                                                     
                                                                                                                               
  2. update() — flat YAML body (no Template wrapper)                                                                           
                            
  When a user passes a simple YAML file via duploctl service apply -f svc.yaml, the body is flat. Before the fix, update() crashed with KeyError: 'Template'.
                                                                                                                               
  ```
# svc.yaml — what the user provides
  Name: my-service                                                                                                             
  Image: nginx:latest
  Replicas: 2 
```                                                                                                                 
                            
  vs. what the API returns (Template-wrapped):                                                                                 
   
  ```
{                                                                                                                            
    "Name": "my-service",   
    "Template": {
      "Image": "nginx:latest",
      "Replicas": 2,                                                                                                           
      "OtherDockerConfig": "{\"Env\":[]}",
      "AgentPlatform": 7                                                                                                       
    }                       
  }
```

  The fix uses body.get("Template", body) and falls back to the existing service state for OtherDockerConfig/AgentPlatform so we don't overwrite them with defaults.
                                                                                                                               
  3. update_env() / update_labels() — empty OtherDockerConfig

  Some services have OtherDockerConfig set to an empty string instead of "{}". Before the fix, loads("") raised  JSONDecodeError.
                                                                                                                               
  ```
{                         
    "Name": "my-service",                                                                                                      
    "Template": {
      "OtherDockerConfig": "",                                                                                                 
      "AllocationTags": ""  
    }
  }
```

  The fix uses .get("OtherDockerConfig") or "{}" to normalize empty/missing values before parsing.

File: src/tests/test_service.py [hunk 1]
```diff
@@ -116,3 +116,89 @@ def test_list_pods(mocker):
...
+  mocker.patch.object(
+    DuploService.__bases__[0], 'find',
+    side_effect=DuploNotFound("test-svc", "Service")
+  )
+  with pytest.raises(DuploNotFound):
+    service.find("test-svc")
+
+
+@pytest.mark.unit
+def test_update_with_flat_yaml_body(mocker):
```
Comment (Matheusbafutto) [inline src/tests/test_service.py:140]: So much better to see the edge cases handled! Thanks for the test coverage :)

**Review by Matheusbafutto — APPROVED** [review]

**Review by kferrone — APPROVED** [review]
lgtm
